### PR TITLE
AVX-14578: Add new resource aviatrix_transit_cloudn_conn

### PR DESCRIPTION
**Pull request by mlin-aviatrix**

### CODE REVIEW

File: aviatrix/resource_aviatrix_transit_cloudn_conn.go [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+			"gw_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the Transit Gateway.",
+			},
+			"insane_mode": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_transit_cloudn_conn.go:48]: Give it a default value. Be consistent with GUI.

File: aviatrix/resource_aviatrix_transit_cloudn_conn.go [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+			"cloudn_neighbor_as_num": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "CloudN LAN Interface Neighbor's BGP ASN.",
+				ValidateFunc: goaviatrix.ValidateASN,
+			},
+			"enable_ha": {
+				Type:        schema.TypeBool,
+				Optional:    true,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_transit_cloudn_conn.go:93]: Give a default value. Be consistent with GUI.

File: aviatrix/resource_aviatrix_transit_cloudn_conn.go [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+				ValidateFunc: goaviatrix.ValidateASN,
+			},
+			"backup_insane_mode": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Enable Insane Mode for connection to Backup Aviatrix CloudN.",
+			},
+			"backup_direct_connect": {
+				Type:        schema.TypeBool,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_transit_cloudn_conn.go:132]: Same.

File: aviatrix/resource_aviatrix_transit_cloudn_conn.go [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+			},
+			"backup_cloudn_neighbor_as_num": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Description:  "Backup Aviatrix CloudN LAN Interface Neighbor's BGP ASN.",
+				ValidateFunc: goaviatrix.ValidateASN,
+			},
+			"backup_insane_mode": {
+				Type:        schema.TypeBool,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_transit_cloudn_conn.go:125]: Same.

File: aviatrix/resource_aviatrix_transit_cloudn_conn.go [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+				Description: "Enable Insane Mode for connection to Backup Aviatrix CloudN.",
+			},
+			"backup_direct_connect": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Enable direct connect to Backup Aviatrix CloudN over private network.",
+			},
+			"enable_load_balancing": {
+				Type:        schema.TypeBool,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_transit_cloudn_conn.go:139]: Same.

File: aviatrix/resource_aviatrix_transit_cloudn_conn.go [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Enable load balancing between Aviatrix CloudN and Backup CloudN.",
+			},
+			"enable_learned_cidrs_approval": {
+				Type:        schema.TypeBool,
+				Optional:    true,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_transit_cloudn_conn.go:147]: Same.

File: docs/resources/aviatrix_transit_cloudn_conn.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+# Create an Aviatrix Transit Gateway to CloudN Connection
+resource "aviatrix_transit_cloudn_conn" "test" {
+  vpc_id                  = "vpc-abcd1234"
+  connection_name         = "my_conn"
+  gw_name                 = "transitGw"
+  bgp_local_as_num        = "123"
+  cloudn_as_num           = "345"
+  cloudn_remote_ip        = "172.12.13.14"
+  cloudn_neighbor_ip      = "182.1.2.3"
+  cloudn_neighbor_as_num  = "65005"
```
Comment (zhiqiangjin) [inline docs/resources/aviatrix_transit_cloudn_conn.md]: One space between "as_num" and "=".

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

**Review by shuoliugit — APPROVED** [review]

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**